src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -56,13 +56,16 @@ export function DiscoverMainRoute(props: Props) {
     dataViewEditor,
   } = services;
   const [error, setError] = useState<Error>();
+  const [loading, setLoading] = useState(true);
   const [savedSearch, setSavedSearch] = useState<SavedSearch>();
-  const dataView = savedSearch?.searchSource?.getField('index');
   const [dataViewList, setDataViewList] = useState<DataViewListItem[]>([]);
   const [hasESData, setHasESData] = useState(false);
   const [hasUserDataView, setHasUserDataView] = useState(false);
   const [showNoDataPage, setShowNoDataPage] = useState<boolean>(false);
   const { id } = useParams<DiscoverLandingParams>();
+  useEffect(() => {
+    setLoading(true);
```

Comment:
I'm curious why this is all by itself. Seems like it could be called with `loadSavedSearch` - but maybe there's something I'm not seeing.

Reply:
yes, you're right, better this way, thx! 👍